src/ds/thread_messaging.h

```diff
@@ -14,10 +14,12 @@ namespace threading
 {
   struct ThreadMsg
   {
-    void (*cb)(std::unique_ptr<ThreadMsg>);
+    void (*cb)(std::unique_ptr<ThreadMsg>) = nullptr;
+    std::function<void()> f = nullptr;
```

Comment:
why is this specialization needed? we should try to minimize the number of ways an async function should be called

Reply:
It isn't. I just wanted to use lambdas for the functions, but I think I ended up not using this at all.